src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -408,36 +370,34 @@ private static string GetFixCategory(DiagnosticSeverity severity)
                 TextSpan? selection = null;
                 if (IsForeground())
                 {
-                    selection = TryGetCodeRefactoringSelection(range);
+                    selection = TryGetCodeRefactoringSelection(state, range);
                 }
                 else
                 {
                     await InvokeBelowInputPriorityAsync(() =>
                     {
-                        // Make sure we were not disposed between kicking off this work and getting
-                        // to this point.
-                        if (IsDisposed)
-                        {
+                        // Make sure we were not disposed between kicking off this work and getting to this point.
+                        using var state = _state.TryAddReference();
+                        if (state is null)
                             return;
-                        }
 
-                        selection = TryGetCodeRefactoringSelection(range);
+                        selection = TryGetCodeRefactoringSelection(state, range);
                     }, cancellationToken).ConfigureAwait(false);
                 }
 
                 return selection;
             }
 
-            private async Task<string?> GetFixLevelAsync(
-                SuggestedActionsSourceProvider provider,
+            private static async Task<string?> GetFixLevelAsync(
+                ReferenceCountedDisposable<State> state,
```

Comment:
same feedback.  would prefer the topmost call to 'own' it's RCD, but then just pass the target along for all of these guys.

Reply:
Objects wrapped in RCD should never be passed in unwrapped form, since it loses information about valid lifetime.

Reply:
hrmm... i was about to be ok with that.  but that seems super odd.  the caller should still own this lifetime, (unless it is truly firing it off blindly).

That said, this may just be my initial apprehensino.  i don't see a problem in passing this along, and if you find value in that, then that's enough.